### PR TITLE
Bulk import of (relevant) docs from devel to 2.3

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY
docs.ansible.com/ansible/ now goes shows you the docs from `stable-2.3`, there are a number of new pages and some changes that are not 2.4 specific that need to be made live (or we end up with broken links)

This needs careful review. There may be other changes that are missing.

##### ISSUE TYPE
 - Docs Pull Request

### CODE REVIEW

**Comment by gundalow** [comment]
There is a reuse page that I've not moved across. not sure about that one.

**Comment by bcoca** [comment]
also need to wait till end of stable-2.3 freeze

File: docs/docsite/rst/dev_guide/developing_module_utilities.rst [hunk 1]
```diff
@@ -9,8 +9,10 @@ Ansible provides a number of module utilities that provide helper functions that
 The following is a list of module_utils files and a general description. The module utility source code lives in the `./lib/module_utils` directory under your main Ansible path - for more details on any specific module utility, please see the source code.
 
 - a10.py - Utilities used by the a10_server module to manage A10 Networks devices.
+- aireos.py - Definitions and helper functions for modules that manage Cisco WLC devices.
```
Comment (bcoca) [inline docs/docsite/rst/dev_guide/developing_module_utilities.rst:12]: most of these files in the new entries only exist in 2.4

File: docs/docsite/rst/dev_guide/developing_modules_checklist.rst [hunk 2]
```diff
@@ -22,8 +22,8 @@ Contributing Modules Checklist
 The following  checklist items are important guidelines for people who want to contribute to the development of modules to Ansible on GitHub. Please read the guidelines before you submit your PR/proposal.
 
 * The shebang must always be ``#!/usr/bin/python``.  This allows ``ansible_python_interpreter`` to work
-* Modules must be written to support Python 2.4. If this is not possible, required minimum python version and rationale should be explained in the requirements section in ``DOCUMENTATION``.  This minimum requirement will be advanced to Python-2.6 in Ansible-2.4.
-* Modules must be written to use proper Python-3 syntax.  At some point in the future we'll come up with rules for running on Python-3 but we're not there yet.  See :doc:`developing_modules_python3` for help on how to do this.
+* Modules must be written to support Python 2.6. If this is not possible, required minimum Python version and rationale should be explained in the requirements section in ``DOCUMENTATION``.  In Ansible-2.3 the minimum requirement for modules was Python-2.4.
```
Comment (bcoca) [inline docs/docsite/rst/dev_guide/developing_modules_checklist.rst:25]: this is incorrect for 2.3, it only applies to 2.4

**Review by bcoca — CHANGES_REQUESTED** [review]
found a couple of issues right away, might be worth splitting this in pr per doc as the whole thing will take very long to review and is not even visible in the github UI

**Comment by gundalow** [comment]
I've started to split this into multiple PRs
`dev_guide`: https://github.com/ansible/ansible/pull/27205
`network`: https://github.com/ansible/ansible/pull/27204

**Comment by dharmabumstead** [comment]
Is this still active?

**Review by dharmabumstead — APPROVED** [review]

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
Is this still an active WIP?

**Comment by gundalow** [comment]
Killing this as we will be always displaying the `devel` fro dev_guide

**Pull request closed without merging**